core/src/main/java/hudson/model/Run.java

```diff
@@ -2410,7 +2410,7 @@ public String getEntryID(Run e) {
 
     private static class DefaultFeedAdapter implements FeedAdapter<Run> {
         public String getEntryTitle(Run entry) {
-            return entry+" ("+entry.getBuildStatusSummary().message+")";
+            return entry.getDisplayName()+" ("+entry.getBuildStatusSummary().message+")";
```

Comment:
entry.getDisplayName() is not localizable. It just a build number or something else user set itself for example by 'currentBuild.displayName' from a pipeline. 

From hudson/models/Run.java
``` 
    @Exported
    public String getDisplayName() {
        return displayName!=null ? displayName : "#"+number;
    }

    public boolean hasCustomDisplayName() {
        return displayName!=null;
    }

    /**
     * @param value
     *      Set to null to revert back to the default "#NNN".
     */
    public void setDisplayName(String value) throws IOException {
        checkPermission(UPDATE);
        this.displayName = value;
        save();
    }
```